packages/side-nav/src/SideNavItem.tsx

```diff
@@ -83,9 +85,11 @@ const activeStyle = css`
   background-color: ${uiColors.green.light3};
 
   &:hover {
+    background-color: ${uiColors.green.light3};
```

Comment:
So instead of setting background color on the button itself, that color should always be on the pseudo-element `:before`. Right now, this is making what was previously a smooth transition into one that looks stepped.